### ISSUE TITLE
Storyshots should allow config object as well as config directory location

### ISSUE THREAD

**Issue by theneva** [storybooks/storybook#3260]
### Issue details

I want to use Storyshots to verify the output of a component using [react-beautiful-dnd](https://github.com/atlassian/react-beautiful-dnd), which requires refs. Since react-test-renderer does not support refs, I'm using Enzyme's `mount` as a renderer (which was my basis for #3252 yesterday).

However, with close to 100 stories to snapshot, it takes pretty much forever to use `mount` for all stories (and ~100 snapshots result in a gigantic (~1MB) `.js.snap` file) in a project that's expected to grow quite quickly (it's a shared CSS and component library for all teams in our company to use and contribute to).

For this reason, I've written two different test files that call `initStoryshots`. The two tests need have equal config, except two things:

1. Which `renderer` to use
2. Which stories are loaded

I can specify which renderer to use (1) as a config option to `initStoryshots`, which is nice.

However, which stories to load must be specified in `config.js` within the config direcrory location specified to `initStoryshots`. That means each separate test (calling `initStoryshots`) requires me to have a separate directory in my project with quasi-duplicated configuration.

Nesting config locations allowed me to get this beast down to something I suppose is okay (`rtr` means `react-test-renderer`):

![image](https://user-images.githubusercontent.com/1404650/37765577-c9391a20-2dc4-11e8-8275-82518338796a.png)

The inner directories' `load-<renderer>-stories.js` files export a function that (when called) requires all the stories related to `<renderer>`. These functions are imported by `stories-<renderer>/config.js` and passed to `@storybook/react`'s `configure(<here>, module)`.

The outer directory is the configuration location of Storybook itself, which sets up addons, custom Webpack configuration for Storysource, and so on. The outer `config.js` requires and calls each `load-<renderer>-stories.js` to add them with the proper wrapping and addons.

It still feels a bit convoluted, though, and requires me to do this:

![image](https://user-images.githubusercontent.com/1404650/37765999-020942f2-2dc6-11e8-963d-61463b10cbcf.png)

It would be much better if I could simply pass a function that loads (requires) stories (and, if needed, some addons config for Storybook and custom Webpack config) to `initStoryshots` instead of passing it a config directory location.

I would also like to be able to split the Storyshots tests into much smaller units (grouped by functionality) to be able to run the tests in parallel, as ~100 tests already takes quite a while. That quickly turns into a mess, as I would need to add new config directories and files and require them for both Storybook itself and each Storyshots test file.

I suppose this issue could (and should?) be resolved by adding parallelism to the Storyshots runner itself, but as it stands, allowing inline configuration to `initStoryshots` instead of configuration directories would help with this, too.

### Steps to reproduce

N/A: this is a feature request.

### Please specify which version of Storybook and optionally any affected addons that you're running

- `@storybook/react 3.3.15`
- `@storybook/addon-storyshots 3.3.15`

### Affected platforms 

- Latest storyshots (3.3.15)

**Comment by igor-dv** [comment]
What about passing a path to the config.js itself? In this case, you will be able to have the following structure:

```
.storybook
 -- config.js
 -- config.specific.js
 -- config.another.js
```

```
initStoryshots({
  configPath: '.storybook/config.specific.js',
});
```

**Comment by igor-dv** [comment]
The next release (v4.0) will be breaking, so we can change things to make an API better. If you have some prototype in mind feel free to share (and PR)

**Comment by theneva** [comment]
I'll have a look at it when I get some time, thanks a lot :)

**Comment by igor-dv** [comment]
Will be able to do with a new `config` prop

**Comment by Hypnosphi** [comment]
Released as [`4.0.0-alpha.10`](https://github.com/storybooks/storybook/releases/tag/v4.0.0-alpha.10)
